inst/support.hpp

```diff
@@ -76,7 +76,7 @@ void user_check_array_dim(cpp11::sexp x, const char *name,
   for (size_t i = 0; i < N; ++i) {
     if (dim[(int)i] != dim_expected[i]) {
       Rf_error("Incorrect size of dimension %d of '%s' (expected %d)",
```

Comment:
```suggestion
      Rf_error("Incorrect size of dimension %zu of '%s' (expected %d)",
```

Given that `i` has type `size_t`, `%zu` should be the portable way of printing it. If you do that then there's no need for a `static_cast`.

Reply:
Alternatively define `i` as an `int` in the `for` loop, then the cast in the line above can also be removed

Reply:
Thanks - I've gone for %zu as the minimal change - I'm not sure the reason for size_t over int here, but let's keep that and just fix the print warning...